node/pkg/lifecycle/startup/startup.go

```diff
@@ -703,6 +703,9 @@ func configureASNumber(node *libapi.Node) bool {
 			return true
 		}
 	} else {
+		if node.Spec.BGP == nil {
```

Comment:
We do probably need this logic right before line 702, though, since we're setting the ASNumber on a potentially nil BGP object.

Reply:
Seems totally fine to me as well, will return early on policy only mode

Reply:
If we fail early we will get the following errors, this seems to be no valid solution

```
2023-06-28 15:58:23.518 [WARNING][65] felix/int_dataplane.go 2061: failed to disable XDP: Failed to wipe the XDP state after 10 tries over 100ms seconds: Error cannot find XDP object "/usr/lib/calico/bpf/filter.o", will proceed anyway.
2023-06-28 15:58:23.954 [WARNING][65] felix/int_dataplane.go 1727: failed to wipe the XDP state error=cannot find XDP object "/usr/lib/calico/bpf/filter.o" try=0
2023-06-28 15:58:24.090 [WARNING][65] felix/int_dataplane.go 1727: failed to wipe the XDP state error=cannot find XDP object "/usr/lib/calico/bpf/filter.o" try=1
2023-06-28 15:58:24.210 [WARNING][65] felix/int_dataplane.go 1727: failed to wipe the XDP state error=cannot find XDP object "/usr/lib/calico/bpf/filter.o" try=2
2023-06-28 15:58:24.338 [WARNING][65] felix/int_dataplane.go 1727: failed to wipe the XDP state error=cannot find XDP object "/usr/lib/calico/bpf/filter.o" try=3
2023-06-28 15:58:24.462 [WARNING][65] felix/int_dataplane.go 1727: failed to wipe the XDP state error=cannot find XDP object "/usr/lib/calico/bpf/filter.o" try=4
2023-06-28 15:58:24.602 [WARNING][65] felix/int_dataplane.go 1727: failed to wipe the XDP state error=cannot find XDP object "/usr/lib/calico/bpf/filter.o" try=5
2023-06-28 15:58:24.738 [WARNING][65] felix/int_dataplane.go 1727: failed to wipe the XDP state error=cannot find XDP object "/usr/lib/calico/bpf/filter.o" try=6
2023-06-28 15:58:24.882 [WARNING][65] felix/int_dataplane.go 1727: failed to wipe the XDP state error=cannot find XDP object "/usr/lib/calico/bpf/filter.o" try=7
2023-06-28 15:58:25.014 [WARNING][65] felix/int_dataplane.go 1727: failed to wipe the XDP state error=cannot find XDP object "/usr/lib/calico/bpf/filter.o" try=8
2023-06-28 15:58:25.150 [WARNING][65] felix/int_dataplane.go 1727: failed to wipe the XDP state error=cannot find XDP object "/usr/lib/calico/bpf/filter.o" try=9
```

Reply:
We are already initializing the BGP data

https://github.com/projectcalico/calico/blob/18c6886a7d766fb112119ab005303c5ffa639c10/node/pkg/lifecycle/startup/startup.go#L485-L491

but not, if it's the policy only mode

https://github.com/projectcalico/calico/blob/18c6886a7d766fb112119ab005303c5ffa639c10/node/pkg/lifecycle/startup/startup.go#L260-L266

Reply:
I will get the same felix errors if I initialize the BGP data pointer right at the beginning of the func `configureASNumber` 🤔

Reply:
I will test more later. I think I have an issue with my local dev env which I have to fix first. Even my working fix does not work anymore if I create a new container 🤔

Reply:
Okay, got it. Had to do a `make clean` before building the new image. Now it's working as expected.